packages/features/bookings/Booker/components/hooks/useBookings.ts

```diff
@@ -0,0 +1,279 @@
+import { useRouter, useSearchParams } from "next/navigation";
+import { useRef, useState } from "react";
+
+import { createPaymentLink } from "@calcom/app-store/stripepayment/lib/client";
+import dayjs from "@calcom/dayjs";
+import { useBookerStore } from "@calcom/features/bookings/Booker/store";
+import type { useEventReturnType } from "@calcom/features/bookings/Booker/utils/event";
+import { updateQueryParam, getQueryParam } from "@calcom/features/bookings/Booker/utils/query-param";
+import {
+  createBooking,
+  createRecurringBooking,
+  mapBookingToMutationInput,
+  mapRecurringBookingToMutationInput,
+  createInstantBooking,
+  useTimePreferences,
+} from "@calcom/features/bookings/lib";
+import { getFullName } from "@calcom/features/form-builder/utils";
+import { useBookingSuccessRedirect } from "@calcom/lib/bookingSuccessRedirect";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { useRouterQuery } from "@calcom/lib/hooks/useRouterQuery";
+import { bookingMetadataSchema } from "@calcom/prisma/zod-utils";
+import { trpc } from "@calcom/trpc";
+import { showToast } from "@calcom/ui";
+
+import type { useBookingFormReturnType } from "./useBookingForm";
+import { useInitialFormValues } from "./useInitialFormValues";
+
+export interface IUseBookings {
+  event: useEventReturnType;
+  hashedLink?: string | null;
+  bookingForm: useBookingFormReturnType["bookingForm"];
+}
+
+export interface IUseBookingLoadingStates {
+  creatingBooking: boolean;
+  creatingRecurringBooking: boolean;
+  creatingInstantBooking: boolean;
+}
+
+export interface IUseBookingErrors {
+  hasDataErrors: boolean;
+  dataErrors: unknown;
+}
+
+export const useBookings = ({ event, hashedLink, bookingForm }: IUseBookings) => {
+  const router = useRouter();
+  const eventSlug = useBookerStore((state) => state.eventSlug);
+  const setFormValues = useBookerStore((state) => state.setFormValues);
+  const rescheduleUid = useBookerStore((state) => state.rescheduleUid);
+  const bookingData = useBookerStore((state) => state.bookingData);
+  const timeslot = useBookerStore((state) => state.selectedTimeslot);
+  const seatedEventData = useBookerStore((state) => state.seatedEventData);
+  const { t, i18n } = useLocale();
+  const bookingSuccessRedirect = useBookingSuccessRedirect();
+  const bookerFormErrorRef = useRef<HTMLDivElement>(null);
+  const [expiryTime, setExpiryTime] = useState<Date | undefined>();
+  const recurringEventCount = useBookerStore((state) => state.recurringEventCount);
+  const isInstantMeeting = useBookerStore((state) => state.isInstantMeeting);
+  const duration = useBookerStore((state) => state.selectedDuration);
+  const { timezone } = useTimePreferences();
+  const username = useBookerStore((state) => state.username);
+  const routerQuery = useRouterQuery();
+  const searchParams = useSearchParams();
+
+  const isRescheduling = !!rescheduleUid && !!bookingData;
+
+  const { initialValues, key } = useInitialFormValues({
```

Comment:
Does the call of useInitialFormValues belongs in this hook? I see that useInitialFormValues is also called in useBookingForm which exports key.

useInitialFormValues seems different than useBookings and key and initialValues that useBookings export are not used, so i think that useInitialFormValues call in this file can be removed.